XenAdmin/Wizards/PatchingWizard/PatchingWizard_AutomatedUpdatesPage.cs

```diff
@@ -506,32 +519,14 @@ private PlanAction GetAfterApplyGuidanceAction(Host host, after_apply_guidance g
 
         private void ShowErrors()
         {
-            if (ErrorMessages != null)
-            {
-                labelTitle.Text = Messages.PATCHINGWIZARD_AUTOUPDATINGPAGE_FAILED;
-                labelError.Text = Messages.PATCHINGWIZARD_AUTOUPDATINGPAGE_ERROR;
-
-                textBoxLog.Text += ErrorMessages;
+            var sb = new StringBuilder();
+            sb.AppendLine();
 
-                log.ErrorFormat("Error message displayed: {0}", labelError.Text);
-                pictureBox1.Image = SystemIcons.Error.ToBitmap();
-                panel1.Visible = true;
-            }
-        }
+            int errorCount = 0;
 
-        private string ErrorMessages
-        {
-            get
+            foreach (var bgw in backgroundWorkers)
             {
-                if (errorActions.Count == 0)
-                    return null;
-
-                var sb = new StringBuilder();
-
-                sb.AppendLine();
-                sb.AppendLine(errorActions.Count > 1 ? Messages.PATCHINGWIZARD_AUTOUPDATINGPAGE_ERRORS_OCCURRED : Messages.PATCHINGWIZARD_AUTOUPDATINGPAGE_ERROR_OCCURRED);
-
-                foreach (var action in errorActions)
+                foreach (var action in bgw.DoneActions)

```

Comment:
As now we are itterating through `DoneActions`, it is expected that `action.Error` is null for non-failed actions, so we shouldm't the log thet the action failed with an empty exception (line 534 below)

Reply:
Yes, I only noticed afterwards, the next commit is fixing this.